### PR TITLE
Fix failed case(except affects related) in flaw_detail_public.feature

### PR DESCRIPTION
**Pull request by RichardXuan**

### CODE REVIEW

**Review by huiwangredhat — COMMENTED** [review]
Met the failed cases:
 Scenario: Modify acknowledgement                          # features/flaw_detail_public.feature:37
    Given I am an analyst AND I am logged into OSIM         # features/steps/common_steps.py:13 21.646s
    And I set the bugzilla api key and jira api key         # features/steps/common_steps.py:18 5.162s
    And I go to a public flaw detail page                   # features/steps/common_steps.py:29 5.433s
    When I edit the first acknowledgement in correct format # features/steps/flaw_detail.py:122 0.184s
      Traceback (most recent call last):
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/behave/model.py", line 1329, in run
          match.run(runner.context)
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/behave/matchers.py", line 98, in run
          self.func(context, *args, **kwargs)
        File "features/steps/flaw_detail.py", line 126, in step_impl
          flaw_detail_page.click_first_ack_edit_btn()
        File "/Users/huiwang/issue-2901/osim/features/pages/flaw_detail_page.py", line 298, in click_first_ack_edit_btn
          self.click_first_edit_btn(self.acknowledgmentCountLabel)
        File "/Users/huiwang/issue-2901/osim/features/pages/flaw_detail_page.py", line 295, in click_first_edit_btn
          self.click_button_with_js(first_ack_edit_btn[0])
      IndexError: list index out of range

    Then Acknowledgement is changed

Scenario: Add external reference                  # features/flaw_detail_public.feature:70
    Given I am an analyst AND I am logged into OSIM # features/steps/common_steps.py:13 13.249s
    And I set the bugzilla api key and jira api key # features/steps/common_steps.py:18 5.731s
    And I go to a public flaw detail page           # features/steps/common_steps.py:29 8.696s
    When I add two external references to the flaw  # features/steps/flaw_detail.py:287 50.370s
    Then Two external references added              # features/steps/flaw_detail.py:297 5.149s
      Traceback (most recent call last):
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/behave/model.py", line 1329, in run
          match.run(runner.context)
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/behave/matchers.py", line 98, in run
          self.func(context, *args, **kwargs)
        File "features/steps/flaw_detail.py", line 303, in step_impl
          flaw_detail_page.check_value_exist(context.second_value)
        File "/Users/huiwang/issue-2901/osim/features/pages/base.py", line 51, in check_value_exist
          self.driver.find_element(By.XPATH, f'//span[contains(text(), "{value}")]')
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/remote/webdriver.py", line 741, in find_element
          return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/remote/webdriver.py", line 347, in execute
          self.error_handler.check_response(response)
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/remote/errorhandler.py", line 229, in check_response
          raise exception_class(message, screen, stacktrace)
      selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //span[contains(text(), "https://test.com/8JJ0100U")]; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
      Stacktrace:
      RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
      WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5
      NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:509:5
      dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16


  Scenario: Add RHSB reference                      # features/flaw_detail_public.feature:74
    Given I am an analyst AND I am logged into OSIM # features/steps/common_steps.py:13 11.484s
    And I set the bugzilla api key and jira api key # features/steps/common_steps.py:18 5.197s
    And I go to a public flaw detail page           # features/steps/common_steps.py:29 5.366s
    When I add two RHSB references to the flaw      # features/steps/flaw_detail.py:306 15.697s
      Traceback (most recent call last):
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/seleniumpagefactory/Pagefactory.py", line 53, in __getattr__
          element = WebDriverWait(self.driver, self.timeout).until(
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/support/wait.py", line 101, in until
          raise TimeoutException(message, screen, stacktrace)
      selenium.common.exceptions.TimeoutException: Message: 
      Stacktrace:
      RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
      WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:191:5
      NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:509:5
      dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16
      
      
      During handling of the above exception, another exception occurred:
      
      Traceback (most recent call last):
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/behave/model.py", line 1329, in run
          match.run(runner.context)
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/behave/matchers.py", line 98, in run
          self.func(context, *args, **kwargs)
        File "features/steps/flaw_detail.py", line 309, in step_impl
          flaw_detail_page.delete_all_reference()
        File "/Users/huiwang/issue-2901/osim/features/pages/flaw_detail_page.py", line 398, in delete_all_reference
          self.wait_msg("referenceDeletedMsg")
        File "/Users/huiwang/issue-2901/osim/features/pages/base.py", line 46, in wait_msg
          element = getattr(self, msg_element)
        File "/Users/huiwang/Library/Python/3.9/lib/python/site-packages/seleniumpagefactory/Pagefactory.py", line 57, in __getattr__
          raise ElementNotFoundException(
      seleniumpagefactory.Pagefactory.ElementNotFoundException: An exception of type TimeoutException occurred. With Element -: referenceDeletedMsg - locator: (xpath, //div[text()='Reference deleted.'])

    Then Only one RHSB reference can be added       # None

File: features/flaw_detail_public.feature [hunk 1]
```diff
@@ -46,8 +46,7 @@ Feature: Flaw detail testing on public flaw
       When I update the random input fields
         |         field |
         |         title |
-        |     component |
-        |      assignee |
```
[the remark this answers is not in the document]
Comment (RichardXuan) [reply features/flaw_detail_public.feature:50]: This field currently do not accept random string, it cover by case " Assign flaw to self"

File: features/pages/flaw_detail_page.py [hunk 2]
```diff
@@ -23,7 +25,7 @@ class FlawDetailPage(BasePage):
 
     def __init__(self, driver):
         self.driver = driver
-        self.timeout = 15
+        self.timeout = 35
```
[the remark this answers is not in the document]
Comment (RichardXuan) [reply features/pages/flaw_detail_page.py:28]: All case works fine in my side. Should we continue increase the time value ?

File: features/flaw_detail_public.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (RichardXuan) [reply features/flaw_detail_public.feature:50]: So which account should we use in this case, anyone is fine?

File: features/pages/flaw_detail_page.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (RichardXuan) [reply features/pages/flaw_detail_page.py:28]: OK

**Comment by huiwangredhat** [comment]
Some failed cases: 

- Then The dropdown source value is updated         # features/steps/flaw_detail.py:114 8.399s

      Assertion Failed: source value should be VENDORSEC, got XEN

- seleniumpagefactory.Pagefactory.ElementNotFoundException: An exception of type TimeoutException occurred. With Element -: acknowledgmentSavedMsg - locator: (xpath, //div[text()="Acknowledgment created."])
-  seleniumpagefactory.Pagefactory.ElementNotFoundException: An exception of type TimeoutException occurred. With Element -: firstAcknowledgementValueText - locator: (xpath, (//div[contains(text(), "from")])[1])

How about changing the timeout to 60s directly?

**Comment by huiwangredhat** [comment]
LGTM.

**Review by huiwangredhat — APPROVED** [review]

**Pull request merged**